cloud/openstack/os_keystone_service.py

```diff
@@ -0,0 +1,210 @@
+#!/usr/bin/python
+# Copyright 2016 Sam Yaple
+#
+# This module is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+try:
+    import shade
+    HAS_SHADE = True
+except ImportError:
+    HAS_SHADE = False
+
+from distutils.version import StrictVersion
+
+DOCUMENTATION = '''
+---
+module: os_keystone_service
+short_description: Manage OpenStack Identity services
+extends_documentation_fragment: openstack
+author: "Sam Yaple (@SamYaple)"
+version_added: "2.2"
+description:
+    - Create, update, or delete OpenStack Identity service. If a service
+      with the supplied name already exists, it will be updated with the
+      new description and enabled attributes.
+options:
+   name:
+     description:
+        - Name of the service
+     required: true
+   description:
+     description:
+        - Description of the service
+     required: false
+     default: None
+   enabled:
+     description:
+        - Is the service enabled
+     required: false
+     default: True
+   type:
```

Comment:
We can, however it is named type in the keystone api (service_type is keystone v2) and this is the service module. I personally have no preference on this matter.